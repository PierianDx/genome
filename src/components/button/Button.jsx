/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './Button.module.scss';

export const Button = ({
  as = 'button',
  size,
  state,
  isStatic,
  rounded,
  outlined,
  inverted,
  fullwidth,
  loading,
  text,
  asDelete,
  className,
  children,
  ref,
  ...props
}) => {
  const classes = cx(className, {
    [`is-${size}`]: size,
    [`is-${state}`]: state,
    'is-static': isStatic,
    'is-rounded': rounded,
    'is-outlined': outlined,
    'is-inverted': inverted,
    'is-fullwidth': fullwidth,
    'is-loading': loading,
    'is-text': text,
    button: !asDelete,
    delete: !!asDelete
  });
  const Element = as;

  return (
    <Element ref={ref} className={classes} {...props}>
      {children}
    </Element>
  );
};

Button.propTypes = {
  as: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  size: PropTypes.oneOf(['small', 'normal', 'medium', 'large']),
  state: PropTypes.oneOf(['hovered', 'focused', 'active', 'disabled']),
  isStatic: PropTypes.bool,
  rounded: PropTypes.bool,
  outlined: PropTypes.bool,
  inverted: PropTypes.bool,
  fullwidth: PropTypes.bool,
  loading: PropTypes.bool,
  text: PropTypes.bool,
  asDelete: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node,
  ref: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
};
Button.defaultProps = {
  as: 'button',
  size: null,
  state: null,
  isStatic: false,
  rounded: false,
  outlined: false,
  inverted: false,
  fullwidth: false,
  loading: false,
  text: false,
  asDelete: false,
  className: '',
  children: null,
  ref: null
};

export default React.forwardRef((props, ref) => <Button ref={ref} {...props} />);
