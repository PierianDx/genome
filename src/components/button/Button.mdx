---
name: Button
---

import { Playground, PropsTable } from "docz";
import Button, { Button as ButtonWithProps } from "./Button";

# Button

This is a `Button` component.

<PropsTable of={ButtonWithProps} />

## Basic Usage

<Playground>
  <div className="buttons">
    <Button>Push Me</Button>
    <Button className="is-primary">Primary Button</Button>
  </div>
</Playground>
