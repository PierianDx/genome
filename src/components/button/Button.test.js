import React from 'react';
import { shallow } from 'enzyme';
import { Button } from './Button';

describe('Button', () => {
  it('Should render', () => {
    const wrapper = shallow(<Button>Push Me</Button>);
    expect(wrapper.text()).toEqual('Push Me');
    expect(wrapper).toMatchSnapshot();
  });
});
