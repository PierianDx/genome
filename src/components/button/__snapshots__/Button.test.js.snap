// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render 1`] = `
<button
  className="button"
>
  Push Me
</button>
`;
